src/providers/ChaptersProvider.tsx

```diff
@@ -0,0 +1,65 @@
+import { Chapter } from '@prisma/client';
+import { createContext, useEffect, useState } from 'react';
+
+export interface ChaptersContextProps {
+  chapters: Record<number, Chapter>;
+  upsertChapter: (data: Chapter) => void;
+  removeChapter: (id: number) => void;
+}
+
+export interface ChaptersProviderProps {
+  children: JSX.Element;
+  initChapters: Chapter[];
+}
+
+export const ChaptersContext = createContext<ChaptersContextProps>({
+  chapters: {},
+  removeChapter: (id: number) => {
+    throw Error('Method not implemented');
+  },
+  upsertChapter: (x: Chapter) => {
+    throw Error('Method not implemented');
+  },
+});
+
+export const ChaptersProvider = ({
+  children,
+  initChapters,
+}: ChaptersProviderProps) => {
+  const [chapters, setChapters] = useState<Record<number, Chapter>>({});
+
+  useEffect(() => {
+    const parsedChapters: Record<number, Chapter> = {};
+    for (let i = 0; i < initChapters.length; i += 1) {
+      const currentChapter = initChapters[i];
+      parsedChapters[currentChapter.id] = currentChapter;
+    }
+    setChapters(parsedChapters);
+  }, [initChapters]);
+
+  const removeChapter = (id: number) => {
+    if (id in chapters) {
+      delete chapters[id];
```

Comment:
Mutating here doesn't look right to me. `setChapters(chapters => chapters.filter(chapter => chapter.id !== id))` perhaps?

Reply:
Yup sorry it was a placeholder code and fixed it in a different branch. I forgot to update this branch with those changes. I have a follow-up branch that handles deletion (with confirmation modal).